Lib/test/test_shutil.py

```diff
@@ -1570,26 +1570,62 @@ def test_tarfile_root_owner(self):
 
     def test_make_archive_cwd(self):
         current_dir = os.getcwd()
-        root_dir = self.mkdtemp()
-        def _breaks(*args, **kw):
+        def archiver(base_name, base_dir, **kw):
+            self.assertNotIn('root_dir', kw)
+            if root_dir is None:
```

Comment:
This is referring to nonlocal variable `root_dir` which is first "defined" by being set far below. That makes for confusing code. Rather than an `if root_dir ...:` conditional within the archiver, it would be cleaner to register and unregister a different archiver for each of the two test scenarios below.